sycl/test/regression/fsycl-host-compiler.cpp

```diff
@@ -0,0 +1,41 @@
+// RUN: %clang++ -fsycl -fsycl-host-compiler=g++ -o %t1.exe %s
```

Comment:
Does it make sense to include `-fsycl-host-compiler-options` option to the E2E tests too?

Reply:
I guess we could - maybe turn on optimizations or something?  I just wanted to cover the basic functionality here, testing that -fsycl-host-compiler-options is doing the right thing is in the driver tests.

Reply:
I added a simple predefined macro setting check that uses `-fsycl-host-compiler-options`